src/gateway/WebSocket.ts

```diff
@@ -3,39 +3,52 @@ import WebSocket from 'ws';
 import { EVENTS, Payload } from './constants';
 import { GATEWAY_OPCODES, GATEWAY_CLOSE_EVENT_CODES } from '../util/Constants';
 
-export default function Socket(token: string, wsURL: string, intents: number) {
-  let interval: NodeJS.Timeout;
-  let lastHeartbeatAck = true;
-  let session_id: string;
-  let seq: number;
-  let ws: WebSocket;
-
-  function heartbeat(expected: boolean) {
+export default class Socket {
+  private _intents: number;
+  private _token: string;
+  private _wsURL: string;
+  private interval!: NodeJS.Timeout;
+  private lastHeartbeatAck = true;
+  private seq!: number;
+  private session_id!: string;
+  private ws!: WebSocket;
+
+  constructor(token: string, wsURL: string, intents: number) {
+    this._token = token;
+    this._wsURL = wsURL;
+    this._intents = intents;
+
+    this.identify = this.identify.bind(this);
+    this.onClose = this.onClose.bind(this);
+    this.onMessage = this.onMessage.bind(this);
+  }
+
+  heartbeat(expected: boolean) {
     if (expected) {
-      if (!lastHeartbeatAck) {
-        clearInterval(interval);
-        ws.removeEventListener('close');
-        ws.terminate();
-
-        newWS();
-        resume();
-        lastHeartbeatAck = true;
-      } else lastHeartbeatAck = false;
+      if (!this.lastHeartbeatAck) {
+        clearInterval(this.interval);
+        this.ws.removeEventListener('close');
+        this.ws.terminate();
+
+        this.newWS();
+        this.resume();
+        this.lastHeartbeatAck = true;
+      } else this.lastHeartbeatAck = false;
```

Comment:
idk what is the configuration in the linter here, I personally prefer to always use new lines and brackets. If that's the choosen configuration we can stick with it, but if that's indifferent I feel like it makes the code clearer and spaced out...